src/terminal.rs

```diff
@@ -165,21 +152,19 @@ impl TextField {
             None => {}
             Some(OpChar::Char(t)) => {
                 match t {
-                    '\u{08}' | '\u{7f}' => {
+                    '\u{08}' => {
                         // backspace
-                        let curs_char = self.cursor.character.char();
-                        self.text
-                            .write(self.cursor.pos.0, self.cursor.pos.1, curs_char);
+                        // 1. Move cursor back 1
+                        // 2. Write a space over the new existing character
                         self.move_cursor(0, -1);
-                        // self.cursor.set_char(' ');
-                        self.text.write(self.cursor.pos.0, self.cursor.pos.1, ' ');
-                    }
-
-                    // these two don't work so hot yet, because of terminal <-> serial interaction reasons
-                    // '\r' => self.cursor.offset(0, -icursor.1),
-                    // '\n' => self.cursor.offset(1, -icursor.1),
+                        self.text.write(self.cursor.pos.row(), self.cursor.pos.col(), ' ');
+                    },
+                    '\u{07}' => {
+                        // Bell not impl
+                    },
 
-                    // taken from char::escape_default (below)
+                    '\u{7f}' => {}
+        
```

Comment:
Might want to add this back in?

```suggestion
         // taken from char::escape_default (below)
```